src/com/facebook/buck/apple/AppleLibraryDescription.java

```diff
@@ -369,6 +375,55 @@ public boolean hasFlavors(ImmutableSet<Flavor> flavors) {
     }
   }
 
+  private <A extends AppleNativeTargetDescriptionArg> ImmutableList<HeaderSymlinkTree>
+  requireHeaderSymlinkTree(
+      BuildRuleParams params,
+      BuildRuleResolver resolver,
+      A args
+  ) {
+    SourcePathResolver pathResolver = new SourcePathResolver(new SourcePathRuleFinder(resolver));
+    CxxLibraryDescription.Arg cxxDelegateArgs = CxxLibraryDescription.createEmptyConstructorArg();
+    AppleDescriptions.populateCxxLibraryDescriptionArg(
+        pathResolver,
+        cxxDelegateArgs,
+        args,
+        params.getBuildTarget());
+
+    ImmutableMap<Path, SourcePath> publicHeaders =  CxxDescriptionEnhancer.parseExportedHeaders(
+        params.getBuildTarget(),
+        pathResolver,
+        Optional.of(defaultCxxPlatform),
+        cxxDelegateArgs);
+    ImmutableMap<Path, SourcePath> privateHeaders =  CxxDescriptionEnhancer.parseHeaders(
+        params.getBuildTarget(),
+        pathResolver,
+        Optional.of(defaultCxxPlatform),
+        cxxDelegateArgs);
+
+    BuildRuleParams untypedParams = params
+        .withoutFlavor(SwiftLibraryDescription.SWIFT_COMPILE_FLAVOR)
+        .withoutFlavor(SwiftLibraryDescription.SWIFT_COMPILE_FLAVOR);
```

Comment:
`SWIFT_COMPANION_FLAVOR`?

Also consider using a FlavorDomain, then you can call `withoutFlavors(flavorDomain.getFlavors())`